app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java

```diff
@@ -0,0 +1,215 @@
+package com.feedhenry.securenativeandroidtemplate.features.device;
+
+import android.app.Activity;
+import android.app.ActivityManager;
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Debug;
+import android.support.v4.content.ContextCompat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.RadioButton;
+
+import com.feedhenry.securenativeandroidtemplate.R;
+import com.feedhenry.securenativeandroidtemplate.features.device.presenters.DevicePresenter;
+import com.feedhenry.securenativeandroidtemplate.features.device.views.DeviceView;
+import com.feedhenry.securenativeandroidtemplate.features.device.views.DeviceViewImpl;
+import com.feedhenry.securenativeandroidtemplate.mvp.views.BaseFragment;
+import com.scottyab.rootbeer.RootBeer;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import dagger.android.AndroidInjection;
+
+import static android.content.Context.ACTIVITY_SERVICE;
+import static android.content.Context.KEYGUARD_SERVICE;
+
+/**
+ * The fragment for the Device related functionality.
+ */
+public class DeviceFragment extends BaseFragment<DevicePresenter, DeviceView> {
+
+
+    public static final String TAG = "device";
+
+    @Inject
+    DevicePresenter devicePresenter;
+
+    @Inject
+    Context context;
+
+    @BindView(R.id.rootAccess)
+    RadioButton rootAccess;
+
+    @BindView(R.id.lockScreenSetup)
+    RadioButton lockScreenSetup;
+
+    @BindView(R.id.emulatorAccess)
+    RadioButton emulatorAccess;
+
+    @BindView(R.id.debuggerAccess)
+    RadioButton debuggerAccess;
+
+    @BindView(R.id.hookingDetected)
+    RadioButton hookingDetected;
+
+    View view;
+
+    public DeviceFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        // Inflate the layout for this fragment
+        view = inflater.inflate(R.layout.fragment_device, container, false);
+        ButterKnife.bind(this, view);
+
+        // perform detections
+        detectRoot();
+        detectDeviceLock();
+        detectEmulator();
+        debuggerDetected();
+        detectHookingFramework();
```

Comment:
what is the hooking framework?

Reply:
Sorry, missed this comment. There are some frameworks that allow you to intercept function calls and replace them with your own - this is where the "hooking" name comes from. Mainly [Xposed](http://repo.xposed.info/module/de.robv.android.xposed.installer) and [Cydia Substrate](http://www.cydiasubstrate.com)